query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -315,6 +322,82 @@ private Filter rebuildFilterOmittingDefaults(
     return builder.clearChildFilter().addAllChildFilter(childFilters).build();
   }
 
+  private QueryRequest rebuildRequest(
+      QueryRequest original,
+      List<Expression> selections,
+      List<Expression> aggregations,
+      Filter filter,
+      List<Expression> groupBys,
+      List<OrderByExpression> orderBys) {
+
+    Set<AttributeExpression> attributeExpressionSet =
+        Stream.concat(
+                findAttributeExpressionFromOrderBy(orderBys).stream(),
```

Comment:
Why are we only doing this for order by and filter? We should do it for group by and selections too (I think I recall one of them wasn't necessary, but consistency is worth it).

Reply:
CC @kotharironak 
I think for others it works without tags__KEYS.

Reply:
I thought it was all of them but group by, maybe selections didn't need it either though. If we're going to pick and choose based on pinot details, that's fine but please leave a comment describing exactly why these two positions are the only ones being treated in this way.

Reply:
resolved